.pre-commit-hooks.yaml

```diff
@@ -13,6 +13,12 @@
   entry: ruby.py
   language: script
   files: \.(rb|rake)$
+- id: restrict_methods_on_reservation_table
+  name: Only allow limited methods on the Reservation model in parsing
+  entry: ^Reservation\s*\.(?!(mass|calendar|hotel_pms_res|set_canceled|update_from)).*
```

Comment:
Good point, I think I got the anchor (`^`) wrong.
However, I don't like the -ve matching for Temporary because we will need to add specifically to the negative list anytime we add a new class with word Reservation.
An easier fix I feel would be using `\b` as the anchor.
That makes the regex: `\bReservation\s*\.(?!(mass|calendar|hotel_pms_res|set_canceled|update_from)).*`
And 
`grep -bzo -P '\bReservation\s*\.(?!(mass|calendar|hotel_pms_res|set_canceled|update_from)).*' /tmp/test.rb`
yields
```
0:Reservation.something
24:Reservation.something
52:Reservation.something% 
```
for me where : 
```
vagrant@vagrant ~/pricematch/admin $% cat /tmp/test.rb
Reservation.something
  Reservation.something
  a = Reservation.something
  TemporaryReservation.something
  Reservation.mass
  TempoReservation.mass
```